### PR TITLE
indexer/golang: add support for methods

### PR DESCRIPTION
**Pull request by joshuabezaleel**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received 
permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

**Comment by joshuabezaleel** [comment]
This is the result on indexing mux package before merging the newly updated UI. 
image

File: doctree/indexer/golang/indexer.go [hunk 1]
```diff
@@ -187,6 +188,67 @@ func (i *goIndexer) IndexDir(ctx context.Context, dir string) (*schema.Index, er
...
+
+				methodDocs := commentsToMarkdown(content, captures["method_docs"])
+				methodName := firstCaptureContentOr(content, captures["method_name"], "")
+				methodReceiver := firstCaptureContentOr(content, captures["method_receiver"], "")
+				methodParams := firstCaptureContentOr(content, captures["method_params"], "")
+				methodResult := firstCaptureContentOr(content, captures["method_result"], "")
+
+				methodType := strings.Split(methodReceiver, " ")[1]
+				methodType = strings.Replace(methodType, "*", "", -1)
+				methodType = strings.Replace(methodType, ")", "", -1)
```
Comment (slimsag) [inline doctree/indexer/golang/indexer.go:229]: I think we should be able to extract the method type name from below `receiver: (parameter_list)` using a better tree-sitter query. Probably this works fine, but that would be the more 'correct' way to do this I think :) I can take a look later, I'd start with printing `captures["method_receiver"]` and seeing what nodes are below it

File: doctree/indexer/golang/indexer.go [hunk 2]
```diff
@@ -261,13 +323,17 @@ func (i *goIndexer) IndexDir(ctx context.Context, dir string) (*schema.Index, er
 				}
 
 				types := typesByPackage[pkgName]
-				types = append(types, schema.Section{
+				typeSchemaSection := schema.Section{
```
Comment (slimsag) [inline doctree/indexer/golang/indexer.go:326]: nit: I think `types` is a better name. there's no need to add the `SchemaSection` suffix. Same with `firstRune` being better than `typeFirstRune` I feel like.

**Review by slimsag — APPROVED** [review]

**Comment by slimsag** [comment]
This looks really fantastic! Thanks for sending this :) I left some ideas for improvement, but I'll go ahead and merge now!

File: doctree/indexer/golang/indexer.go (hunk 1 above)
Comment (joshuabezaleel) [reply doctree/indexer/golang/indexer.go:229]: You're right, I should've been more thorough in checking the query result and its child node. Thank you for pointing this out Stephen! Made a PR for this one 🙂 
https://github.com/sourcegraph/doctree/pull/40

File: doctree/indexer/golang/indexer.go (hunk 2 above)
Comment (joshuabezaleel) [reply doctree/indexer/golang/indexer.go:326]: Thinking about that as well. Changed the `typeFirstRune` but for `types` the variable name was already reserved for [this one](https://github.com/sourcegraph/doctree/blob/main/doctree/indexer/golang/indexer.go#L325) with the type of `[]schema.Section` already and we definitely can't use the name `type` for variable name thus me using `typeSchemaSection`. Do you have recommendation on a more fitting name?

**Pull request merged**